PriceCalculater/IDiscountService.cs

```diff
@@ -3,22 +3,26 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
-namespace PriceCalculater
-{
-    public   interface IDiscountService
+namespace PriceCalculater.Services;
+    public  interface IDiscountService
     {
-        public decimal GetDiscountPercentage();
+        public List<Discount> GetDiscountPercentage(Product product);
     }
     public class DiscountService : IDiscountService
-    {
-        private readonly decimal _percentage;
-        public DiscountService(decimal percentage)
-        {
-            this._percentage = percentage;
-        }
-        public decimal GetDiscountPercentage()
+   { 
+        private readonly decimal _universalpercentage;
+        private readonly Dictionary<long, decimal> _UpcDiscountList;
+      public DiscountService(decimal universalPercentage , Dictionary<long, decimal> UpcDiscountList)
```

Comment:
1- Discount service for now accept dictonary and univversal discount to create needed discount